### DISCUSSION TITLE
Pytest - Creating classes/methods vs modules or functions

### DISCUSSION THREAD

**Discussion by sramasub** [pytest-dev/pytest#11465] [General]
I'm just starting to use pytest for our application testing needs. I would like to know if there's a preferred way of writing tests - either as classes and methods (or) as individual test functions (called as modules, i guess)

What are the pros and cons of it? We don't own or write the code of the application we are testing. We only test its functionality.

**Comment by RonnyPfannschmidt** [comment]
In pytest Test classes are typically only used for code organization

With fixtures, there's hardly any use for classes these days
